packages/app/src/hooks/useSubmitExtrinsic/index.tsx

```diff
@@ -295,13 +299,47 @@ export const useSubmitExtrinsic = ({
     })
   }
 
-  const onError = (type?: string) => {
+  const onError = (type?: string, errorMessage?: string) => {
     if (type === 'ledger') {
       handleResetLedgerTask()
     }
+
+    // Get current balance info for context
+    const { transferrableBalance } = getTransferOptions(from)
+    const txFee = getTxSubmission(uid)?.fee || 0n
+    const hasInsufficientFunds = transferrableBalance < txFee
+
+    let title = t('cancelled')
+    let subtitle = t('transactionCancelled')
+
+    // Provide contextual messages based on cancellation reason
+    if (type === 'insufficient_funds' || hasInsufficientFunds) {
+      title = t('insufficientFunds')
+      subtitle = t('addMoreDotForFees', { unit })
+
+      // Add transaction-specific context
+      if (tx?.call.pallet === 'Staking') {
+        subtitle = t('addMoreDotForStaking', { unit, minAmount: unit })
+      } else if (tx?.call.pallet === 'NominationPools') {
+        subtitle = t('addMoreDotForPooling', { unit })
+      }
+    } else if (
+      type === 'user_cancelled' ||
+      errorMessage?.includes('User rejected') ||
+      errorMessage?.includes('Cancelled') ||
+      errorMessage?.includes('user rejected') ||
+      errorMessage?.includes('cancelled by user')
```

Comment:
Could this logic refer to a utility in `TxSubmission` too that this and `handleError` can use?